.travis.yml

```diff
@@ -0,0 +1,28 @@
+language: cpp
+compiler:
+  - gcc
+  - clang
+
+before_install:
+  - if [ "${TRAVIS_OS_NAME}" = "linux" ]; then
+         wget http://www.cmake.org/files/v3.3/cmake-3.3.0-Linux-i386.tar.gz
+      && tar -xzf cmake-3.3.0-Linux-i386.tar.gz
+      && sudo cp -fR cmake-3.3.0-Linux-i386/* /usr
+      && sudo apt-add-repository -y ppa:ubuntu-toolchain-r/test
+      && sudo apt-get -qq update
+      && sudo apt-get -qq install g++-6 libc6-i386
+      && export CXX="g++-6"
+      && export CC="gcc-6"
```

Comment:
I feel like it would be nicer to pull out installation of dependencies into a separate script, this means you could also test that the commands work in a VM for example.

As for exporting the compiler environment variables, is there a reason for wanting to use gcc 6 over the default installed gcc? If its not a hard requirement then I think it seems nicer to rely on the built in travis handling of the multiple compilers.